src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -102,18 +112,60 @@ private String getHIUId(RequestedDetail requestedDetail) {
     public Mono<ConsentApprovalResponse> approveConsent(String patientId,
                                                         String requestId,
                                                         List<GrantedConsent> grantedConsents) {
-        //TODO validate CareContexts for the patient
         return validatePatient(patientId)
                 .then(validateConsentRequest(requestId))
                 .flatMap(consentRequest ->
-                        Flux.fromIterable(grantedConsents)
-                                .flatMap(grantedConsent -> {
-                                    var consentArtefact = from(consentRequest, grantedConsent);
-                                    String consentArtefactSignature = getConsentArtefactSignature(consentArtefact);
-                                    return consentArtefactRepository.addConsentArtefactAndUpdateStatus(consentArtefact, requestId, patientId, consentArtefactSignature)
-                                            .thenReturn(ConsentArtefactReference.builder().id(consentArtefact.getConsentId()).build());
-                                }).collectList())
-                .map(consents -> ConsentApprovalResponse.builder().consents(consents).build());
+                        generateConsentArtefacts(requestId, grantedConsents, patientId, consentRequest))
+                .flatMap(consents -> {
+                    ConsentApprovalResponse consentApprovalResponse = ConsentApprovalResponse.builder()
+                            .consents(consents)
+                            .build();
+                    return broadcastConsentArtefacts("callBackUrl", consentApprovalResponse.getConsents())
+                            .thenReturn(consentApprovalResponse);
+                });
```

Comment:
we could abstract out this part by having a class or something, PostConsentApproval, then consent manager.class doesn't need to know about amqb and all that.